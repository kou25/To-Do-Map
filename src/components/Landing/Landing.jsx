import React from 'react'
import {Link} from 'react-router-dom'


function Landing() {
    return (
        <div className="container-fluid" id="img" style={{fontFamily:"cursive"}}>
             <Link to='/add' >
        <button data-testid="button" type="submit" style={{ 
        marginTop:"12%", marginLeft:"45%"}}
        className="btn btn-warning shadow">
        <span style={{color:"white", textDecoration:"none",textShadow:"1px 1px 2px black"}}>Get Started</span></button></Link>
        <h1 data-testid="heading" style={{color:"#8B4513", 
        fontFamily:"cursive",
         fontWeight:"bold",
         paddingTop:"45vh",
         paddingLeft:"50px",
         fontSize:"3rem",
         textShadow:"1px 1px 2px black, 0 0 25px white, 0 0 5px white"}}>
             Create <span style={{color:"orange"}}>your own</span> To-Dos...</h1>
        </div>
    )
}

export default Landing
