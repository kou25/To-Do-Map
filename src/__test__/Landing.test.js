import React from 'react';
import ReactDom from 'react-dom'
import Landing from '../components/Landing/Landing'
import { BrowserRouter } from 'react-router-dom';
import {render, cleanup} from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'

afterEach(cleanup);

it("render without crashing",()=>{
  const div= document.createElement("div");
  ReactDom.render(
  <BrowserRouter>
  <Landing/>
  </BrowserRouter>,div);
})

it("renders button correctly", ()=>{
  const{getByTestId}=render(
  <BrowserRouter>
    <Landing/>
    </BrowserRouter>)
  expect(getByTestId('button')).toHaveTextContent('Get Started')
})
it("renders heading correctly", ()=>{
  const{getByTestId}=render(
  <BrowserRouter>
    <Landing/>
    </BrowserRouter>)
  expect(getByTestId('heading')).toHaveTextContent('Create your own To-Dos')
})

it('matches snapshot', ()=>{
  const tree=renderer.create(
  <BrowserRouter>
    <Landing/>
    </BrowserRouter>).toJSON()
  expect(tree).toMatchSnapshot()
})
