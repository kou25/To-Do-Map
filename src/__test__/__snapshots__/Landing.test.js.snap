// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container-fluid"
  id="img"
  style={
    Object {
      "fontFamily": "cursive",
    }
  }
>
  <a
    href="/add"
    onClick={[Function]}
  >
    <button
      className="btn btn-warning shadow"
      data-testid="button"
      style={
        Object {
          "marginLeft": "45%",
          "marginTop": "12%",
        }
      }
      type="submit"
    >
      <span
        style={
          Object {
            "color": "white",
            "textDecoration": "none",
            "textShadow": "1px 1px 2px black",
          }
        }
      >
        Get Started
      </span>
    </button>
  </a>
  <h1
    data-testid="heading"
    style={
      Object {
        "color": "#8B4513",
        "fontFamily": "cursive",
        "fontSize": "3rem",
        "fontWeight": "bold",
        "paddingLeft": "50px",
        "paddingTop": "45vh",
        "textShadow": "1px 1px 2px black, 0 0 25px white, 0 0 5px white",
      }
    }
  >
    Create 
    <span
      style={
        Object {
          "color": "orange",
        }
      }
    >
      your own
    </span>
     To-Dos...
  </h1>
</div>
`;
